overrides/cf.d.ts

```diff
@@ -289,97 +297,693 @@ interface RequestInitCfPropertiesImageMinify {
 }
 
 /**
- * In addition to the properties on the standard Request object,
- * the cf object contains extra information about the request provided
- * by Cloudflare's edge.
- *
- * Note: Currently, settings in the cf object cannot be accessed in the
- * playground.
+ * Request metadata provided by Cloudflare's edge.
  */
-interface IncomingRequestCfProperties {
+type IncomingRequestCfProperties = {
   /**
-   * (e.g. 395747)
+   * [ASN](https://www.iana.org/assignments/as-numbers/as-numbers.xhtml) of the incoming request.
+   *
+   * @example 395747
    */
   asn: number;
+
   /**
-   * The organisation which owns the ASN of the incoming request.
-   * (e.g. Google Cloud)
+   * The organization which owns the ASN of the incoming request.
+   *
+   * @example "Google Cloud"
    */
   asOrganization: string;
+
+  /**
+   * Only set when using Cloudflare Bot Management.
+   */
   botManagement?: IncomingRequestCfPropertiesBotManagement;
```

Comment:
Do we want to encode this optionality in the type system somehow? Something like 
```ts
type IncomingRequestCfProperties = BaseIncomingRequestCfProperties & BotManagementProperties & CloudflareForSaasProperties & GeographicProperties ...
```

You could then have (to solve the other issue with geographic properties):

```ts
type GeographicProperties = OptionA | OptionB | OptionC
```

Reply:
👍